src/neo/SmartContract/Helper.cs

```diff
@@ -272,12 +280,15 @@ public static bool VerifyWitnesses(this IVerifiable verifiable, DataCache snapsh
         internal static bool VerifyWitness(this IVerifiable verifiable, DataCache snapshot, UInt160 hash, Witness witness, long gas, out long fee)
         {
             fee = 0;
+            Script invocationScript = witness.InvocationScript;
+            if (!Check(invocationScript)) return false;
+            Script verificationScript = witness.VerificationScript;
```

Comment:
I cached the `Script` class because it already contains instructions parsed cache